src/main/java/org/opensearch/flowframework/workflow/ProcessNode.java

```diff
@@ -139,51 +139,45 @@ public TimeValue nodeTimeout() {
      * @return this node's future.
      * This is returned immediately, while process execution continues asynchronously.
      */
-    public CompletableFuture<WorkflowData> execute() {
+    public PlainActionFuture<WorkflowData> execute() {
         if (this.future.isDone()) {
             throw new IllegalStateException("Process Node [" + this.id + "] already executed.");
         }
 
         CompletableFuture.runAsync(() -> {
-            List<CompletableFuture<WorkflowData>> predFutures = predecessors.stream().map(p -> p.future()).collect(Collectors.toList());
             try {
-                if (!predecessors.isEmpty()) {
-                    CompletableFuture<Void> waitForPredecessors = CompletableFuture.allOf(predFutures.toArray(new CompletableFuture<?>[0]));
-                    waitForPredecessors.join();
-                }
-
                 logger.info("Starting {}.", this.id);
```

Comment:
Already approved but one quick note. This log entry should be after the `actionGet()` call so it doesn't post until the previous nodes have all finished.